Sources/Web3Core/Utility/Data+Extension.swift

```diff
@@ -33,32 +34,33 @@ extension Data {
         return difference == UInt8(0x00)
     }
 
-    public static func zero(_ data: inout Data) {
+    static func zero(_ data: inout Data) {
         let count = data.count
         data.withUnsafeMutableBytes { (body: UnsafeMutableRawBufferPointer) in
             body.baseAddress?.assumingMemoryBound(to: UInt8.self).initialize(repeating: 0, count: count)
         }
     }
 
-    public static func randomBytes(length: Int) -> Data? {
-        for _ in 0...1024 {
-            var data = Data(repeating: 0, count: length)
-            let result = data.withUnsafeMutableBytes { (body: UnsafeMutableRawBufferPointer) -> Int32? in
-                if let bodyAddress = body.baseAddress, body.count > 0 {
-                    let pointer = bodyAddress.assumingMemoryBound(to: UInt8.self)
-                    return SecRandomCopyBytes(kSecRandomDefault, length, pointer)
-                } else {
-                    return nil
-                }
-            }
-            if let notNilResult = result, notNilResult == errSecSuccess {
-                return data
-            }
+    static func randomBytes(length: Int) -> Data? {
+        let entropy_bit_size = length//128
+        //# valid_entropy_bit_sizes = [128, 160, 192, 224, 256], count: [12, 15, 18, 21, 24]
+        var entropy_bytes = [UInt8](repeating: 0, count: entropy_bit_size)// / 8)
+
+        let status = SecRandomCopyBytes(kSecRandomDefault, entropy_bytes.count, &entropy_bytes)
+
+        if status != errSecSuccess { // Always test the status.
+
+        } else {
+            entropy_bytes = [UInt8](repeating: 0, count: entropy_bit_size)// / 8)
+            arc4random_buf(&entropy_bytes, entropy_bytes.count)
```

Comment:
I'm not sure we will use `arc4random_buf`. I'm concerned about it not being secure enough.

Reply:
I do not think we will allow that. For now `SecRandomCopyBytes` looks like the only option.
Note, that if it fails the most likely reason is that there's something wrong with the machine or it's memory where this function was invoked: https://developer.apple.com/documentation/security/errsecsuccess

Reply:
See this:

> * Here the stream cipher has been modified always to include the time
> * when initializing the state.  That makes it impossible to
> * regenerate the same random sequence twice, so this can't be used
> * for encryption, but will generate good random numbers.

https://opensource.apple.com/source/Libc/Libc-825.26/gen/FreeBSD/arc4random.c.auto.html